### PR TITLE
drivers: caam: make use of generic memalign() implementation

### PR DESCRIPTION
**Pull request by clementfaure**
Make use of the newly implemented memalign() function for the CAAM
driver.
Remove the previous CAAM memalign() implementation and its associated
debugging structures.

Signed-off-by: Clement Faure <clement.faure@nxp.com>

### CODE REVIEW

File: core/drivers/crypto/caam/utils/utils_mem.c [hunk 1]
```diff
@@ -203,13 +77,13 @@ static void *mem_alloc(size_t size, uint8_t type)
```
Comment (jforissier) [inline core/drivers/crypto/caam/utils/utils_mem.c:77]: Perhaps "(normal)" is no more meaningful? (same at other places below)

File: core/drivers/crypto/caam/utils/utils_mem.c [hunk 2]
```diff
@@ -203,13 +77,13 @@ static void *mem_alloc(size_t size, uint8_t type)
 
 	if (type & MEM_TYPE_ALIGN) {
 		cacheline_size = read_cacheline_size();
-		if (ADD_OVERFLOW(alloc_size,
-				 ROUNDUP(alloc_size, cacheline_size),
-				 &alloc_size))
+		if (ROUNDUP_OVERFLOW(alloc_size, cacheline_size, &alloc_size))
```
Comment (jforissier) [inline core/drivers/crypto/caam/utils/utils_mem.c:80]: Is this `ROUNDUP_OVERFLOW()` needed?

Comment (jenswi-linaro) [reply core/drivers/crypto/caam/utils/utils_mem.c:80]: `memalign()` should be OK with any size.

File: core/drivers/crypto/caam/utils/utils_mem.c (hunk 1 above)
Comment (clementfaure) [reply core/drivers/crypto/caam/utils/utils_mem.c:77]: I will remove it

File: core/drivers/crypto/caam/utils/utils_mem.c (hunk 2 above)
Comment (clementfaure) [reply core/drivers/crypto/caam/utils/utils_mem.c:80]: I will discard it

**Comment by jforissier** [comment]
`Acked-by: Jerome Forissier <jerome@forissier.org>`

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by clementfaure** [comment]
I'm gonna force push another patch to discard ```size_t alloc_size = size;``` which is not needed anymore.

**Comment by clementfaure** [comment]
Force push:
 * Remove the ```alloc_size```
 * Rebase to make IBART happy

**Pull request merged**